Microsoft.NET.Build.Containers/ImageConfig.cs

```diff
@@ -29,53 +37,108 @@ internal ImageConfig(JsonNode config)
         _labels = GetLabels();
         _exposedPorts = GetExposedPorts();
         _environmentVariables = GetEnvironmentVariables();
+        _rootFsLayers = GetRootFileSystemLayers();
+        _architecture = GetArchitecture();
+        _os = GetOs();
+        _history = GetHistory();
     }
 
+    private List<HistoryEntry> GetHistory() => _config["history"]?.AsArray().Select(node => node.Deserialize<HistoryEntry>()!).ToList() ?? new List<HistoryEntry>();
+    private string GetOs() => _config["os"]?.ToString() ?? throw new ArgumentException("Base image configuration should contain an 'os' property.");
+    private string GetArchitecture() => _config["architecture"]?.ToString() ?? throw new ArgumentException("Base image configuration should contain an 'architecture' property.");
+
     /// <summary>
     /// Builds in additional configuration and returns updated image configuration in JSON format as string.
     /// </summary>
     internal string BuildConfig()
     {
-        if (_config["config"] is not JsonObject config)
+        var newConfig = new JsonObject();
+
+        if (_exposedPorts.Any())
         {
-            throw new InvalidOperationException("Base image configuration should contain a 'config' node.");
+            newConfig["ExposedPorts"] = CreatePortMap();
+        }
+        if (_labels.Any())
+        {
+            newConfig["Labels"] = CreateLabelMap();
+        }
+        if (_environmentVariables.Any())
+        {
+            newConfig["Env"] = CreateEnvironmentVariablesMapping();
         }
```

Comment:
With an entirely new object, there's no need to even write these keys if there's no data to write.